ql/src/java/org/apache/hadoop/hive/ql/io/orc/RecordReaderImpl.java

```diff
@@ -165,7 +165,9 @@ public boolean nextBatch(VectorizedRowBatch theirBatch) throws IOException {
   public void close() throws IOException {
     super.close();
     // free the memory for the column vectors
-    batch.cols = null;
+    if (batch != null) {
```

Comment:
It is fine. Personally I like more the following syntax, it is bit more readable to me. (You do **not** need to change) 
```
if(batch == null)
   return;
batch.cols = null
```